### ISSUE TITLE
[BUG]  Problem connecting to device

### ISSUE THREAD

**Issue by lucas-linvix** [WhiskeySockets/Baileys#1039]
**Describe the bug**
Hello everyone, I'm having a problem with the Baileys connection with the cell phone I'm connected to. I am currently using version 6.7.7 and my cell phone is connected using WhatsApp Business. 
In the first few days there were no problems, everything was working ok, however, recently I have been having problems, as my connection is active, but I am not able to forward any messages or receive messages via Baileys. For everything to work normally again, I need to disconnect the device and connect it again, however, this behavior is showing instability and I haven't found anything on my consoles or errors regarding this, including for other customers who are also using my code where I connect to your devices, this problem does not occur.

 - Is this on a server?
 
 -> Response:  yes, I'm using DigitalOcean
 
 - What do your `connectOptions` look like? 
 - > Response: 
![image](https://github.com/user-attachments/assets/b127fda9-8038-45ae-b0b4-f8eadf6bfe81)

 - Do you have multiple clients on the same IP?
 -> Response:  yes
 
 - Are you using a proxy?
 ->  Response: No

**Comment by cvchelles** [comment]
Conseguiu Resolver?

**Comment by lucas-linvix** [comment]
> Conseguiu Resolver?
No, I'm still waiting for some help

**Comment by wilsinho8** [comment]
its a problem with libsignal (protocol that decrypts and encrypts the sessions and messages)
no solution yet

**Comment by lucas-linvix** [comment]
> its a problem with libsignal (protocol that decrypts and encrypts the sessions and messages) no solution yet

I understand, do you know if somebody find a solution and make a PR about?

**Comment by Zain-ul-din** [comment]
> its a problem with libsignal (protocol that decrypts and encrypts the sessions and messages) no solution yet

Also, can you guys move the libsignal dependency to HTTP instead of git many people with no SSH are facing errors while installing the package.

**Comment by lucas-linvix** [comment]
> > its a problem with libsignal (protocol that decrypts and encrypts the sessions and messages) no solution yet
> 
> Also, can you guys move the libsignal dependency to HTTP instead of git many people with no SSH are facing errors while installing the package.

By doing this, could it at least help for now to prevent this error from occurring? Did not result in other impacts to other Baileys features

**Comment by lucas-linvix** [comment]
> > > its a problem with libsignal (protocol that decrypts and encrypts the sessions and messages) no solution yet
> > 
> > 
> > Also, can you guys move the libsignal dependency to HTTP instead of git many people with no SSH are facing errors while installing the package.
> 
> @Zain-ul-din  By doing this, could it at least help for now to prevent this error from occurring? Did not result in other impacts to other Baileys features

**Comment by wilsinho8** [comment]
> atures?

will not solve, for now no solution to the problem

**Comment by lucas-linvix** [comment]
> > atures?
> 
> will not solve, for now no solution to the problem

ok, if somebody have any update about a solution or treatment for this error, please share here.

**Comment by Zain-ul-din** [comment]
> > > its a problem with libsignal (protocol that decrypts and encrypts the sessions and messages) no solution yet
> > 
> > 
> > Also, can you guys move the libsignal dependency to HTTP instead of git many people with no SSH are facing errors while installing the package.
> 
> @Zain-ul-din By doing this, could it at least help for now to prevent this error from occurring? Did not result in other impacts to other Baileys features?

yes! All you have to do just change the git dependency to HTTP that's it

**Comment by lucas-linvix** [comment]
hell, now we have just to wait and give good luck to @krub
![image](https://github.com/user-attachments/assets/70d8d3ad-d5ce-435f-b3fe-d88cdacc2edd)

**Comment by wilsinho8** [comment]
> hell, now we have just to wait and give good luck to @krub ![image](https://private-user-images.githubusercontent.com/146751746/371192966-70d8d3ad-d5ce-435f-b3fe-d88cdacc2edd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczNjIzMDIsIm5iZiI6MTcyNzM2MjAwMiwicGF0aCI6Ii8xNDY3NTE3NDYvMzcxMTkyOTY2LTcwZDhkM2FkLWQ1Y2UtNDM1Zi1iM2ZlLWQ4OGNkYWNjMmVkZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNlQxNDQ2NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZjZkZTE1MTRmMWIyZGIyMzk5MjRjMjAwZWNhMmI3MTFhMDE4MDgwNWRmMDRiMzgxYzg4NmYzOTJlMTdiOGUxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.OyJDoGYuil3WE8uqPBxVIkz6t_s7oj8_K8bmLfPhjPg)

Purp too

https://github.com/WhiskeySockets/Baileys/issues/886#issuecomment-2254153101

**Comment by Skidy89** [comment]
purp too (btw im krub)

**Comment by lucas-linvix** [comment]
ok a don`t know who will fix this, but a really grateful if someone can fix this current problem. If you can tell me when, it will be great too. I don't want to boring, but its sothing have with a lot of headache. Again, thanks guys for respond.

**Comment by devora371** [comment]
Using version 6.7.4, I am able to receive messages, but it only works for one WhatsApp account at a time. If I add another account, it don't recive messages for new account. I first need to remove the first device and then add the new one to start receiving messages from the new WhatsApp account. Receiving messages from multiple WhatsApp accounts simultaneously is not possible.

    const { version, isLatest } = await fetchLatestBaileysVersion()
    const WA = makeWASocket({
      version: version,
      printQRInTerminal: true,
      mobile: false,
      keepAliveIntervalMs: 10000,
      syncFullHistory: false,
      downloadHistory: false,
      markOnlineOnConnect: true,
      defaultQueryTimeoutMs: undefined,
      linkPreviewImageThumbnailWidth: 1980,
      generateHighQualityLinkPreview: true,
      auth: state,
      logger: P({ level: 'fatal' }),
      browser: ["JMD Infotech", "Safari", "16.5"],
    });

**Comment by wilsinho8** [comment]
> purp too (btw im krub)

Hey Krub any news my friend?

**Comment by lucas-linvix** [comment]
> > purp too (btw im krub)
> 
> Hey Krub any news my friend?

not yet I have to talk in the community and nobody have a treat ou news about that.

![image](https://github.com/user-attachments/assets/00a2cbb6-7a4b-4cde-98ec-d35be79799d7)

![image](https://github.com/user-attachments/assets/db8cd3e0-3cb0-42fa-ae1b-f7ccc5e9a893)

**Comment by lucas-linvix** [comment]
hello, I hava talk to @Rajeh, he says this problem has been resolved and I already make my tests, so now a just need update my project and see if not have any problem in production.  So, depending on how it goes I will be sending it here, thanks again for your help guys.

![image](https://github.com/user-attachments/assets/ab3914e1-7130-4a19-b024-b47812d36c59)
